### PR TITLE
Intel packages: new versions

### PR DESCRIPTION
**Pull request by robert-mijakovic**
Adds hashes for new versions of Intel packages.

### CODE REVIEW

**Comment by robert-mijakovic** [comment]
@adamjstewart

**Comment by robert-mijakovic** [comment]
I forgot to add the new version of Intel compiler and Intel MPI.
Now it's there.

**Comment by jrood-nrel** [comment]
While we're at it, it looks like there is a Parallel Studio 2019 compiler update 8 as well.

**Comment by robert-mijakovic** [comment]
> While we're at it, it looks like there is a Parallel Studio 2019 compiler update 8 as well.

Hi @jrood-nrel. I cannot find the archive. I searched through the list of available archives on https://registrationcenter.intel.com/en/products/download/3803/ but it is not there.
I only can find it mentioned here: https://software.intel.com/content/www/ru/ru/develop/articles/intel-compiler-and-composer-update-version-numbers-to-compiler-version-number-mapping.html

Searching for the previous release is successful but searching for the update 6, parallel_studio_xe_2019_update6_composer_edition.tgz, is not.

If you know it's location, I'll be happy to add it.

**Comment by jrood-nrel** [comment]
Interesting. I saw it here https://software.intel.com/content/www/us/en/develop/articles/intel-c-compiler-190-for-linux-release-notes-for-intel-parallel-studio-xe-2019.html#history . I seem to be unable to download any of the files on the computer I am using at the moment so I am unsure what the custom URL is for the update 8.

File: var/spack/repos/builtin/packages/intel-mpi/package.py [hunk 1]
```diff
@@ -11,6 +11,8 @@ class IntelMpi(IntelPackage):
 
     homepage = "https://software.intel.com/en-us/intel-mpi-library"
 
+    version('2019.8.217', sha256='fa163b4b79bd1b7509980c3e7ad81b354fc281a92f9cf2469bf4d323899567c0',
```
Comment (jrood-nrel) [inline var/spack/repos/builtin/packages/intel-mpi/package.py]: I think this should be `2020.2.254`.

Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/intel-mpi/package.py]: AFAIK there is no Intel MPI 2020.2.254. Parallel Studio 2020.2 is distributing Intel MPI 2019.8.
IntelMPI2019 8

The next Intel MPI seems to be 2021 instead. Intel MPI versions are 2017, 2018, 2019 and 2021.
https://software.intel.com/content/www/us/en/develop/articles/intel-mpi-library-release-notes.html

Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/intel-mpi/package.py]: Hmm, I downloaded and installed this Intel MPI version and I swear it was 2020.2.254 in the path structure.

Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/intel-mpi/package.py]: I guess it should at least be `2019.8.254`?

Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/intel-mpi/package.py]: Yes, you are right. I have forgotten to update it to 254.

Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/intel-mpi/package.py]: Ok after looking further into this, the versions make sense to me now.

**Comment by jrood-nrel** [comment]
These all look good to me. @adamjstewart would be quite convenient to have these merged for what I'm working on right now 🙂

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@jrood-nrel _ask and ye shall receive_

Sorry for the delay in reviewing, my internet was down for 7 days.

**Pull request merged**